### ISSUE TITLE
Tuples and variance, resulting in type load failure

### ISSUE THREAD

**Issue by svick** [dotnet/roslyn#13705]
**Version Used**: 2.0.0-beta5-60909-06 (latest from roslyn-master-nightly MyGet)

**Steps to Reproduce**:

Compile the following code:

```c#
using System;

namespace System
{
    public struct ValueTuple<T1, T2>
    {
        public ValueTuple(T1 item1, T2 item2) : this() {}
        
        public T1 Item1;
        public T2 Item2;
	}
}

public interface IAsyncEnumerator<out T> {
    ValueTuple<bool, T> TryGetNext();
}

class Program
{
    static void Main()
    {
        Console.WriteLine(typeof(IAsyncEnumerator<>));
    }
}
```

**Expected Behavior**:

Compilation fails, probably with the following error:

> error CS1961: Invalid variance: The type parameter 'T' must be invariantly valid on 'IAsyncEnumerator<T>.TryGetNext()'. 'T' is covariant.

**Actual Behavior**:

Compilation succeeds, but the resulting assembly fails peverify:

```
[token  0x02000002] Type load failed.
[IL]: Error: [… : Program::Main][offset 0x00000001] Unable to resolve token.
```

The assembly also fails at runtime:

```
Unhandled Exception: System.TypeLoadException: Could not load type 'IAsyncEnumerator`1' from assembly … because a covariant or contravariant type parameter was used illegally in the signature for the result type in method 'TryGetNext'.
   at Program.Main()
```

---

The same behavior happens when the `ValueTuple` is used using tuple syntax:

```c#
public interface IAsyncEnumerator<out T> {
    (bool, T) TryGetNext();
}
```

On the other hand, when `ValueTuple` is renamed, compilation correctly fails:

```c#
namespace System
{
    public struct ValueTupl<T1, T2>
    {
        public ValueTupl(T1 item1, T2 item2) : this() {}
        
        public T1 Item1;
        public T2 Item2;
	}
}

public interface IAsyncEnumerator<out T> {
    ValueTupl<bool, T> TryGetNext();
}
```

> (16,5): error CS1961: Invalid variance: The type parameter 'T' must be invariantly valid on 'IAsyncEnumerator<T>.TryGetNext()'. 'T' is covariant.

**Comment by jaredpar** [comment]
Nice bug!  Thanks for reporting.

**Comment by jcouv** [comment]
Thanks a lot for the bug, @svick 
It was a breeze to repro and I have a fix pending.

**Comment by jcouv** [comment]
Fixed by #13729
